src/core/symbology/qgscategorizedsymbolrenderer.cpp

```diff
@@ -956,3 +957,65 @@ QgsDataDefinedSizeLegend *QgsCategorizedSymbolRenderer::dataDefinedSizeLegend()
 {
   return mDataDefinedSizeLegend.get();
 }
+
+int QgsCategorizedSymbolRenderer::matchToSymbols( QgsStyle *style, const QgsSymbol::SymbolType type, QVariantList &unmatchedCategories, QStringList &unmatchedSymbols, const bool caseSensitive, const bool useTolerantMatch )
+{
+  if ( !style )
```

Comment:
Would it be useful to rely on the user's saved styles when no QgsStyle object is provided instead? I don't have any particular scenario in mind, just throwing the idea at you.

Reply:
That's a good idea!